playwright-tests/test-assets/constructed-styles/index.js

```diff
@@ -1,10 +1,41 @@
-// Create a new constructed stylesheet
-const sheet = new CSSStyleSheet();
-sheet.replaceSync(`
+// Create a new constructed stylesheet that uses insertRule and deleteRule.
+const injectSheet = new CSSStyleSheet();
+injectSheet.insertRule(`
+  :root {
+    --my-custom-font: 400 1rem / 1.5rem Roboto;
+    --my-custom-font-weight: 400;
+  }`);
+injectSheet.insertRule(`
   p {
-    color: red;
-  }
-`);
+    font: var(--my-custom-font);
+    font-weight: var(--my-custom-font-weight);
+  }`);
+injectSheet.insertRule('b { color: green; }', 0);
+injectSheet.insertRule('i { color: red; }', 1);
+injectSheet.deleteRule(1); // Remove "i { color: red; }"
```

Comment:
The toEqual assertions to make it explicitly clear that this was removed in the end. Maybe we should add an assertion that says the styles string does not contain this style?

Reply:
Good idea